internal/csi-addons/server/server.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+
+	"google.golang.org/grpc"
+
+	"github.com/ceph/ceph-csi/internal/util/log"
+)
+
+var ErrNoUDS = errors.New("no UNIX domain socket")
+
+// CSIAddonsService is the interface that is required to be implemented so that
+// the CSIAddonsServer can register the service by calling RegisterService().
+type CSIAddonsService interface {
+	// RegisterService is called by the CSIAddonsServer to add a CSI-Addons
+	// service that can handle requests.
+	RegisterService(server grpc.ServiceRegistrar)
+}
+
+// CSIAddonsServer is the gRPC server that listens on an endpoint (UNIX domain
+// socket) where the CSI-Addons requests come in.
+type CSIAddonsServer struct {
+	// URL components to listen on the UNIX domain socket
+	scheme string
+	path   string
+
+	// state of the CSIAddonsServer
```

Comment:
below field does not carry any  `state`  in general.. so wondering are there better explanation on below fields. :)